deltachat-ios/Controller/WebxdcViewController.swift

```diff
@@ -0,0 +1,263 @@
+import UIKit
+import WebKit
+import DcCore
+
+class WebxdcViewController: WebViewViewController {
+
+    enum WebxdcHandler: String {
+        case getStatusUpdates = "getStatusUpdatesHandler"
+        case sendStatusUpdate = "sendStatusUpdateHandler"
+    }
+    let INTERNALSCHEMA = "webxdc"
+    let INTERNALDOMAIN = "local.app"
+
+    var messageId: Int
+    var dcContext: DcContext
+    var webxdcUpdateObserver: NSObjectProtocol?
+
+
+    // Block just everything :)
+    let blockRules = """
+    [
+        {
+            "trigger": {
+                "url-filter": ".*"
+            },
+            "action": {
+                "type": "block"
+            }
+        }
+    ]
+    """
+
+    lazy var webxdcbridge: String = {
+        let script = """
+        window.webxdc = (() => {
+          var update_listener = () => {};
+
+          // instead of calling .getStatusUpdatesHandler (-> async),
+          // we're passing the updates directly to this js function
+          window.__webxdcUpdate = (updateString) => {
+            var updates = JSON.parse(updateString);
+            if (updates.length === 1) {
+              update_listener(updates[0]);
+            }
+          };
+
+          return {
+            selfAddr: "\(dcContext.addr ?? "unknown")",
+
+            selfName: "\(dcContext.displayname ?? "unknown")",
```

Comment:
same here, please escape the user input, though here it is even more important then on the email